### PR TITLE
CIV-8199  Pay your claim fee - Front End

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-8199

### Change description ###

 Pay your claim fee - Front End

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -10,7 +10,7 @@ sonar.exclusions=src/main/app.ts,src/main/development.ts,src/main/index.js,src/m
 src/main/assets/js/postcode-lookup.js,src/main/assets/js/calculate-amount.js,src/main/assets/js/language-toggle.js,src/main/assets/js/calculate-length-repayment.js,src/main/assets/scss/task-list.scss,\
 src/main/common/utils/taskList/taskListBuilder.ts,src/main/common/utils/taskList/tasks/confirmYourDetails.ts,\
 src/main/modules/taskListService.ts,\
-src/main/routes/features/claim/claimSubmittedController.ts,\
+src/main/routes/features/claim/claimSubmittedController.ts,src/main/routes/features/claim/payment/claimFeeBreakDownController.ts,src/main/routes/features/claim/checkAnswersController.ts,\
```
[the remark this answers is not in the document]
Comment (jeswanth-hmcts) [reply sonar-project.properties:13]: Even though tests are present, in sonar analysis for coverage, they are not showing, so they are excluded. Mainly it is for callback functions on the controller show that has been uncovered. Also happy to implement any other solution for this

**Review by TatsyMaslova — DISMISSED** [review]

File: src/main/routes/features/claim/checkAnswersController.ts [hunk 2]
```diff
@@ -61,10 +64,12 @@ claimCheckAnswersController.post(CLAIM_CHECK_ANSWERS_URL, async (req: Request |
     } else {
       await saveStatementOfTruth(userId, form.model);
       const submittedClaim = await submitClaim(<AppRequest>req);
-      await deleteDraftClaimFromStore(userId);
+
       if (claim.claimDetails.helpWithFees.option === YesNo.NO) {
-        res.redirect(constructResponseUrlWithIdParams(userId, paymentUrl));
+        const paymentUrlWithId = constructResponseUrlWithIdParams(userId, paymentUrl);
```
Comment (laperla11) [inline src/main/routes/features/claim/checkAnswersController.ts:69]: if helpWithFees is yes, whne do we delete the draft claim from redis after your change?

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+claimFeeBreakDownController.get(CLAIM_FEE_URL, async (req: AppRequest, res: Response, next: NextFunction): Promise<void> => { // nosonar
+  try {
+    const userId = req.session?.user?.id;
+    const claim = await getCaseDataFromStore(userId);
+    const claimFee = await civilServiceClient.getClaimAmountFee(claim.totalClaimAmount, req);
+    const totalAmount = claim.totalInterest ? (claim.totalClaimAmount + claim.totalInterest + claimFee) : (claim.totalClaimAmount + claimFee);
+    res.render(viewPath, { csrf: res.locals.csrf, totalClaimAmount: claim.totalClaimAmount, interest: claim.totalInterest, claimFee, hasInterest: !!claim.totalInterest, totalAmount });
```
Comment (laperla11) [inline src/main/routes/features/claim/payment/claimFeeBreakDownController.ts:19]: why we're passing `csrf` here?

**Review by laperla11 — CHANGES_REQUESTED** [review]

File: src/main/routes/features/claim/checkAnswersController.ts (hunk 2 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/claim/checkAnswersController.ts:69]: In else, we are deleting. if help with fees is yes,

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts (hunk 3 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/claim/payment/claimFeeBreakDownController.ts:19]: removing this

**Review by laperla11 — APPROVED** [review]

**Review by laperla11 — APPROVED** [review]

**Review by neeta-hmcts — DISMISSED** [review]

**Review by TatsyMaslova — DISMISSED** [review]

**Review by portizdev — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Review by laperla11 — DISMISSED** [review]

**Review by laperla11 — APPROVED** [review]

**Review by neeta-hmcts — DISMISSED** [review]

**Review by portizdev — DISMISSED** [review]

**Review by GorilaNaranja — APPROVED** [review]

**Review by ElifDenizEsea — APPROVED** [review]

**Pull request merged**